src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -89,7 +89,8 @@ std::shared_ptr<AbstractOperator> LQPTranslator::translate_node(const std::share
     return operator_iter->second;
   }
 
-  const auto pqp = _translate_by_node_type(node->type, node);
+  auto pqp = _translate_by_node_type(node->type, node);
```

Comment:
What about IndexScans? I believe those are the only ones without a 1:1 mapping to LQP nodes.

Reply:
Depending on the scan type, we create either a table scan or an index scan. But that shouldn't make a difference here, does it?

Reply:
We always create a table scan, have a look at `_translate_predicate_node_to_index_scan`.

Reply:
Added LQP node to both scans, added an explaining comment, and a test that checks that all three PQP nodes share the same LQP node.